src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpointImpl.java

```diff
@@ -105,7 +105,15 @@ public String getChildEntities(Request request, Response response) {
     public String getEventTypesOfEntity(Request request, Response response) {
         long entityId = getEntityId(request);
 
-		List<EventType> eventTypes = PersistenceAdapterImpl.getInstance().getEventTypes(entityId);
+        EntityHierarchyNode entityNode = HierarchyBuilderImpl.getInstance().getEntityHierarchyRootNode().findChildEntity(entityId);
+
+        if (entityNode == null) {
+        	halt(HttpStatusCodes.NOT_FOUND, "cannot find entity in hierarchy");
+		}
+
+		List<Long> entityIds = entityNode.getChildIds();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'entityNode' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)